packages/contracts/contracts/optimistic-ethereum/ovm/SafetyChecker.sol

```diff
@@ -90,67 +89,30 @@ contract SafetyChecker is ContractResolver {
                     return false;
                 }
                 // append this opcode to a list of ops
-                ops <<= 8;
-                ops |= op;
-                // [0x57, 0x56, 0x00, 0xfd, 0xfe, 0xf3, 0xf1] all have handlers
-                if (opBit & 0x600a00000000000000000000000000000000000000c000000000000000000001 != 0) {
-                    // JUMPI
-                    if (op == 0x57) {
-                        // We can now reach all JUMPDESTs
-                        seenJUMP = true;
-                    // JUMP
-                    } else if (op == 0x56) {
-                        // We can now reach all JUMPDESTs
-                        seenJUMP = true;
-                        // we are now inside unreachable code until we hit a JUMPDEST!
-                        insideUnreachableCode = true;
-                    // STOP or REVERT or INVALID or RETURN (see safety checker docs in wiki for more info)
-                    } else if (op == 0x00 || op == 0xfd || op == 0xfe || op == 0xf3) {
-                        // If we can't jump to JUMPDESTs, then all remaining bytecode is unreachable
-                        if (!seenJUMP) {
-                            return true;
-                        }
+                // [STOP(0x00),JUMP(0x56),RETURN(0xf3),REVERT(0xfd),INVALID(0xfe),CALLER(0x33)] all have handlers
+                if (opBit & 0x6008000000000000000000000000000000000000004000000008000000000001 != 0) {
+                    // STOP or JUMP or RETURN or REVERT or INVALID (see safety checker docs in wiki for more info)
+                    if (opBit & 0x6008000000000000000000000000000000000000004000000000000000000001 != 0) {
                         // We are now inside unreachable code until we hit a JUMPDEST!
                         insideUnreachableCode = true;
                     // CALL
-                    } else if (op == 0xf1) {
-                        // Minimum 4 total ops:
-                        // 1. PUSH1 value
-                        // 2. PUSH20 execution manager address
-                        // 3. PUSH or DUP gas
+                    } else if (op == 0x33) {
```

Comment:
I wouldn't do it like this. CALLER isn't a blocked opcode afaik, and if it's used in other contexts, it will be wrong. Why not check for CALL and look back, like the old contract did?